fuels-signers/src/wallet/mod.rs

```diff
@@ -83,8 +100,144 @@ impl Wallet {
+    ///
+    ///   // Create the actual wallets/signers
+    ///   let wallet_1 = LocalWallet::new_from_private_key(pk_1, provider.clone()).unwrap();
+    ///   let wallet_2 = LocalWallet::new_from_private_key(pk_2, provider).unwrap();
+    ///
+    ///   // Transfer 1 from wallet 1 to wallet 2
+    ///   let _receipts = wallet_1
+    ///        .transfer(&wallet_2.address(), 1, Default::default())
+    ///        .await
+    ///        .unwrap();
+    ///
+    ///   let wallet_2_final_coins = wallet_2.get_coins().await.unwrap();
+    ///
+    ///   // Check that wallet two now has two coins
+    ///   assert_eq!(wallet_2_final_coins.len(), 2);
+    ///   Ok(())
+    /// }
+    /// ```
+    pub async fn transfer(
+        &self,
+        to: &Address,
+        amount: u64,
+        color: Color,
+    ) -> io::Result<Vec<Receipt>> {
+        let spendable = self.get_spendable_coins(&color, amount).await?;
+
+        let mut inputs: Vec<Input> = vec![];
+        let mut outputs: Vec<Output> = vec![];
+
+        let mut amount_to_send = 0;
+
+        let output_coin = Output::coin(*to, amount, color);
+        outputs.push(output_coin);
+
+        for coin in spendable {
+            if amount_to_send >= amount {
+                break;
+            }
+
+            let input_coin = Input::coin(
+                UtxoId::from(coin.utxo_id),
+                coin.owner.into(),
+                coin.amount.0,
+                color,
+                0,
+                0,
+                vec![],
+                vec![],
+            );
+
+            inputs.push(input_coin);
+
+            amount_to_send += coin.amount.0;
+        }
+
+        // Calculate change, if applicable
+        if amount_to_send > amount {
+            let change = Output::change(self.address(), amount_to_send - amount, color);
+            outputs.push(change);
+        }
+
+        // Build transaction and sign it
+        let mut tx = self.provider.build_transfer_tx(&inputs, &outputs);
+        let sig = self.sign_transaction(&tx).await.unwrap();
+        self.append_signatures(&mut tx, &[sig]);
+
+        // Note that currently coins being sent aren't marked as spent by the client.
+        // This will be coming up soon.
+        self.provider.send_transaction(&tx).await.map(Into::into)
+    }
+
+    /// Gets coins from this wallet
+    /// Note that this is a simple wrapper on provider's `get_coins`.
+    pub async fn get_coins(&self) -> Result<Vec<Coin>, WalletError> {
+        Ok(self.provider.get_coins(&self.address()).await?)
+    }
+
+    /// Gets spendable coins from this wallet.
+    /// Note that this is a simple wrapper on provider's
+    /// `get_spendable_coins`.
+    pub async fn get_spendable_coins(&self, color: &Color, amount: u64) -> io::Result<Vec<Coin>> {
+        Ok(self
+            .provider
+            .get_spendable_coins(&self.address(), *color, amount)
+            .await?)
+    }
 }
 
 #[cfg_attr(target_arch = "wasm32", async_trait(?Send))]
```

Comment:
`transfer` should probably have unit tests too, since it is a non-trivial process ( checking the amount, change …)

Reply:
https://github.com/FuelLabs/fuels-rs/pull/110/files#diff-f3c02edc378fe2fe43610a513ab6b6d5cd57233e86d1794f2f564030ce385d50R178-R216